Source: http://bugs.mysql.com/bug.php?id=52514
Upstream: Yes, will be in 5.1.50
Reason: ABI check fails with gcc 4.5

From: Date: July 20 2010 7:44pm
Subject: bzr commit into mysql-5.1-bugteam branch (davi:3493) Bug#52514
List-Archive: http://lists.mysql.com/commits/113968
X-Bug: 52514

 3493 Davi Arnaut	2010-07-20
      Bug#52514: mysql 5.1 do_abi_check does not compile w/ gcc4.5
                 due to GCC preprocessor change
            
      The problem is that newer GCC versions treats missing headers
      as fatal errors. The solution is to use a guard macro to prevent
      the inclusion of system headers when checking the ABI with the
      C Preprocessor.
      
      Reference: http://gcc.gnu.org/bugzilla/show_bug.cgi?id=15638
                 http://gcc.gnu.org/bugzilla/show_bug.cgi?id=44836
     @ Makefile.am
        Define guard macro.
     @ include/mysql.h
        Guard the header inclusion.
     @ include/mysql.h.pp
        Header is not included anymore.

    modified:
      Makefile.am
      include/mysql.h
      include/mysql.h.pp
--- a/Makefile.am	2010-05-12 10:29:02 +0000
+++ b/Makefile.am	2010-07-20 17:44:29 +0000
@@ -314,7 +314,7 @@ abi_check_all:	$(TEST_PREPROCESSOR_HEADE
 do_abi_check:
 	set -ex; \
 	for file in $(abi_headers); do \
-	         @CC@ -E -nostdinc -dI \
+	         @CC@ -E -nostdinc -dI -DMYSQL_ABI_CHECK \
 	                  -I$(top_srcdir)/include \
 	                  -I$(top_srcdir)/include/mysql \
 	                  -I$(top_srcdir)/sql \
--- a/include/mysql.h	2009-11-24 21:25:23 +0000
+++ b/include/mysql.h	2010-07-20 17:44:29 +0000
@@ -44,7 +44,9 @@ extern "C" {
 #endif
 
 #ifndef _global_h				/* If not standard header */
+#ifndef MYSQL_ABI_CHECK
 #include <sys/types.h>
+#endif
 #ifdef __LCC__
 #include <winsock2.h>				/* For windows */
 #endif
--- a/include/mysql.h.pp	2009-11-24 21:25:23 +0000
+++ b/include/mysql.h.pp	2010-07-20 17:44:29 +0000
@@ -1,4 +1,3 @@
-#include <sys/types.h>
 typedef char my_bool;
 typedef int my_socket;
 #include "mysql_version.h"
